packages/react-components/src/components/xy-output-component.tsx

```diff
@@ -151,6 +156,39 @@ export class XYOutputComponent extends AbstractTreeOutputComponent<AbstractOutpu
             ;
     }
 
+    renderYAxis(): React.ReactNode {
+        // Y axis with D3
+        const chartHeight = parseInt(this.props.style.height.toString());
+        const margin = { top: 15, right: 0, bottom: 5, left: this.props.style.yAxisWidth };
+        let allMax = 0;
+        this.state.xyData?.datasets?.forEach((dSet: any) => {
+            const rowMax = Math.max(...dSet.data);
+            allMax = Math.max(allMax, rowMax);
+        });
+        const yScale = scaleLinear()
+            .domain([0, Math.max(allMax, 1)])
+            .range([chartHeight - margin.bottom, margin.top]);
+        const yTransform = `translate(${margin.left}, 0)`;
+        // Abbreviate large numbers
+        const scaleYLabel = (d: number) => (
+            d >= 1000000000000 ? d / 1000000000000 + 'G' :
+            d >= 1000000000 ? d / 1000000000  + 'B':
+            d >= 1000000 ? d / 1000000 + 'M' :
+            d >= 1000 ? d / 1000 + 'K':
+            d
+        );
+        if (allMax > 0) {
+            select(this.yAxisRef.current).call(axisLeft(yScale).tickSizeOuter(0).ticks(4)).call(g => g.select('.domain').remove());
```

Comment:
Below a certain height we might want to have only 2 ticks? Otherwise the tick labels overlap.

Also, we use ticks(4) when we want 5 ticks. Not cool. Not your fault :(

Reply:
In D3, the value of "ticks" is not guaranteed. The scale makes an effort to achieve that number of divisions, taking into consideration other limitations, like avoiding overlapping (see https://observablehq.com/@d3/d3-ticks).
It is possible to achieve an exact division of the scale with a custom implementation, but "ticks" provides a fairly good solution already.
In case you find we really need a custom solution, I suggest it to be done in a separate ticket.

Reply:
OK!